bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/protocol/QbusCommunication.java

```diff
@@ -0,0 +1,926 @@
+                logger.info("Connected via local port {} from thread {}", socket.getLocalPort(),
+                        Thread.currentThread().getId());
+            } else {
+                return;
+            }
+
+            setSN();
+            getSN();
+
+            // Connect to Qbus server
+            Connect();
+
+            // If Qbus Client is connected then initialize, else put Bridge offline
+            if (CTDConnected == true) {
+                initialize();
+                (new Thread(qEvents)).start();
+            } else {
+                handler.bridgeOffline("No communication with Qbus client");
+            }
+        } catch (IOException | InterruptedException e) {
+            logger.warn("Error initializing communication from thread {}", Thread.currentThread().getId());
+            // No connection with Qbus server, put Bridge offline
+            stopCommunication();
+            if (handler != null) {
+                handler.bridgeOffline("No communication with Qbus server");
+            }
+
+        }
+    }
+
+    /**
+     * Cleanup socket when the communication with Qbus Server is closed.
+     *
+     * @throws IOException
+     *
+     */
+    public synchronized void stopCommunication() {
+        listenerStopped = true;
+
+        Socket socket = qSocket;
+
+        if (socket != null) {
+            try {
+                socket.close();
+            } catch (IOException ignore) {
+                // ignore IO Error when trying to close the socket if the intention is to close it anyway
+            }
+        }
+
+        qSocket = null;
+
+        CTDConnected = false;
+        logger.debug("Communication stopped from thread {}", Thread.currentThread().getId());
+    }
+
+    /**
+     * Close and restart communication with Qbus Server.
+     */
+    public synchronized void restartCommunication() {
+        stopCommunication();
+
+        logger.debug("Qbus: restart communication from thread {}", Thread.currentThread().getId());
+
+        startCommunication();
+    }
+
+    /**
+     * Method to check if communication with Qbus Server is active
+     *
+     * @return True if active
+     */
+    public boolean communicationActive() {
+
+        return qSocket != null;
+    }
+
+    /**
+     * Method to check if communication with Qbus Client is active
+     *
+     * @return True if active
+     */
+
+    public boolean clientConnected() {
+        return CTDConnected;
+    }
+
+    /**
+     * Runnable that handles incomming communication from Qbus server.
+     * <p>
+     * The thread listens to the TCP socket opened at instantiation of the {@link QbusCommunication} class
+     * and interprets all incomming json messages. It triggers state updates for active channels linked to the
+     * Qbus outputs. It is started after initialization of the communication.
+     *
+     */
+    private Runnable qEvents = () -> {
```

Comment:
You could make this a private method. Then, you can use it when creating the thread by `this::myMethod`.
```suggestion
    private void myMethod() {
```